src/ntrip_client/rtcm_parser.py

```diff
@@ -64,19 +64,22 @@ def parse(self, buffer):
     rtcm_packets = []
     while index < len(combined_buffer):
       # Find the start of the RTCM 3.2 packet
-      if combined_buffer[index] == _RTCM_3_2_PREAMBLE:
+      ascii_values = []
```

Comment:
For me the buffer still comes in as a byte array by default, and this causes the node to crash for me. so we should have some logic in here to handle both cases. Also I think that you could use the `encode` function instead of looping. I had success with the following implementation, but as I said, my buffer is already bytes, so please let me know if this works for you

```python
if isinstance(combined_buffer, str):
  ascii_values = combined_buffer.encode('utf-8')
elif isinstance(combined_buffer, bytes):
  ascii_values = combined_buffer
else:
  raise Exception('Buffer type is "{}" not a string or bytes, so we cannot parse it.'.format(type(combined_buffer)))
```

Also I think a better name for `ascii_values` would be `combined_buffer_bytes`